str_analysis/convert_strling_calls_to_expansion_hunter_json.py

```diff
@@ -0,0 +1,179 @@
+"""
+
+import argparse
+import collections
+import json
+import os
+import re
+import pandas as pd
+
+
+"""
+ExpansionHunter output format:
+
+  "LocusResults": {
+        "chr12-57610122-57610131-GCA": {
+          "AlleleCount": 2,
+          "Coverage": 50.469442942130875,
+          "FragmentLength": 433,
+          "LocusId": "chr12-57610122-57610131-GCA",
+          "ReadLength": 151,
+          "Variants": {
+            "chr12-57610122-57610131-GCA": {
+              "CountsOfFlankingReads": "(1, 1), (2, 4)",
+              "CountsOfInrepeatReads": "()",
+              "CountsOfSpanningReads": "(2, 1), (3, 48), (6, 1)",
+              "Genotype": "3/3",
+              "GenotypeConfidenceInterval": "3-3/3-3",
+              "ReferenceRegion": "chr12:57610122-57610131",
+              "RepeatUnit": "GCA",
+              "VariantId": "chr12-57610122-57610131-GCA",
+              "VariantSubtype": "Repeat",
+              "VariantType": "Repeat"
+            }
+          }
+        },
+
+  "SampleParameters": {
+        "SampleId": "NA19239",
+        "Sex": "Female"
+  }
+
+"""
+
+"""
+STRling txt calls format:
+
+$1                    #chrom : chr11
+$2                      left : 119648275
+$3                     right : 119648276
+$4                repeatunit : GGT
+$5               allele1_est : 2.00
+$6               allele2_est : 22.03
+$7            anchored_reads : 3
+$8            spanning_reads : 14
+$9            spanning_pairs : 8
+$10  expected_spanning_pairs : 68.42
+$11      spanning_pairs_pctl : 0.00
+$12               left_clips : 0
+$13              right_clips : 0
+$14           unplaced_pairs : 0
+$15                    depth : 34.0
+$16           sum_str_counts : 126
+"""
+
+
+def main():
+    p = argparse.ArgumentParser()
+    p.add_argument("strling_txt_path", nargs="+", help="STRling calls .txt path(s)")
+    args = p.parse_args()
+
+    for strling_txt_path in args.strling_txt_path:
+        print(f"Processing {strling_txt_path}")
+        locus_results = process_strling_txt(strling_txt_path)
+
+        output_json_path = re.sub(".txt$", "", strling_txt_path) + ".json"
+        print(f"Writing results for", len(locus_results["LocusResults"]), f"loci to {output_json_path}")
+        with open(output_json_path, "wt") as f:
+            json.dump(locus_results, f, indent=3)
+
+
+def compute_CI(allele_size, repeat_unit_length):
+    """Compute a pseudo "confidence interval" based on the criteria using for mendelian violation tests in the
+    STRling pre-print - where a genotype was considered not to be a mendelian violation if it was within
+    25% or 10bp of the expected genotype.
+
+    Args:
+        allele_size (int): The number of repeats.
+        repeat_unit_length (str): Length of the repeat unit in base pairs.
+
+    Returns:
+         str: "{lower}-{upper}" representing the lower/upper bound in numbers of repeats.
+    """
+    max_repeats_distance = 10.0/repeat_unit_length
```

Comment:
Wow, thank you for tracking that down! 
I was basing this on the wording in the [STRling preprint](https://www.biorxiv.org/content/10.1101/2021.11.18.469113v1.full) 
"STRling alleles were considered matched if their sizes were within 25% of the parent allele or ten bp." 
but agree the code uses 10 repeats rather than 10bp.